plugins/modules/consul_agent_check.py

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, Michael Ilg
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: consul_agent_check
+short_description: Add, modify & delete checks within a consul cluster
+description:
+ - Registers checks for an agent with a consul cluster.
+   A service is some process running on the agent node that should be advertised by
+   consul's discovery mechanism. It may optionally supply a check definition,
+   a periodic service test to notify the consul cluster of service's health.
+ - "Checks may also be registered per node e.g. disk usage, or cpu usage and
+   notify the health of the entire node to the cluster.
+   Service level checks do not require a check name or id as these are derived
+   by Consul from the Service name and id respectively by appending 'service:'
+   Node level checks require a O(check_name) and optionally a O(check_id)."
+ - Currently, there is no complete way to retrieve the script, interval or TTL
+   metadata for a registered check. Without this metadata it is not possible to
+   tell if the data supplied with ansible represents a change to a check. As a
+   result this does not attempt to determine changes and will always report a
+   changed occurred. An API method is planned to supply this metadata so at that
+   stage change management will be added.
+ - "See U(http://consul.io) for more details."
+author: "Michael Ilg (@Ilgmi)"
+extends_documentation_fragment:
+  - community.general.consul
+attributes:
+  check_mode:
+    support: none
```

Comment:
It would be great if you could try to implement at least check mode. Modules should always try to support check mode. (Usually this is easy since in most cases you need to determine whether something is changed first anyway, before doing the change. At that point you already know what would happen in check mode.)

Reply:
Both check_mode and diff_mode basically come for free with ConsulModule anyways. Might need some small adjustments here and there but should work mostly painless.